Polyfills/Canvas/Source/Image.cpp

```diff
@@ -65,13 +79,42 @@ namespace Babylon::Polyfills::Internal
         return Napi::Value::From(Env(), m_src);
     }
 
-    void NativeCanvasImage::SetSrc(const Napi::CallbackInfo&, const Napi::Value& value)
+    void NativeCanvasImage::SetSrc(const Napi::CallbackInfo& info, const Napi::Value& value)
     {
         auto text{value.As<Napi::String>().Utf8Value()};
-        m_src = text;
+        UrlLib::UrlRequest request{};
+        request.Open(UrlLib::UrlMethod::Get, text);
+        request.ResponseType(UrlLib::UrlResponseType::Buffer);
+        request.SendAsync().then(m_runtimeScheduler, arcana::cancellation::none(), [env{ info.Env() }, this, request{std::move(request)}](arcana::expected<void, std::exception_ptr> result) {
+            if (result.has_error())
+            {
+                throw Napi::Error::New(env, result.error());
+            }
+
+            Dispose();
+
+            auto buffer = request.ResponseBuffer();
+            m_imageContainer = bimg::imageParse(&m_allocator, buffer.data(), static_cast<uint32_t>(buffer.size_bytes()));
+            if (m_imageContainer == nullptr)
+            {
+                throw Napi::Error::New(env, "Unable to decode image with provided src for in Canvas.");
+            }
+
+            m_width = m_imageContainer->m_width;
+            m_height = m_imageContainer->m_height;
+
+            m_onloadHandlerRefs.Call({});
+        });
+    }
+
+    void NativeCanvasImage::SetOnload(const Napi::CallbackInfo&, const Napi::Value& value)
+    {
+        Napi::Function eventHandler = value.As<Napi::Function>();
+        m_onloadHandlerRefs = Napi::Persistent(eventHandler);
```

Comment:
The name for this variable is a bit odd since there is only one ref.

```suggestion
        m_onloadHandlerRef = Napi::Persistent(eventHandler);
```